adafruit_bitmapsaver.py

```
@@ -0,0 +1,120 @@
+    output_file.write(struct.pack('<I', bitmap.height))
+    output_file.write(struct.pack('<H', 1))
+    output_file.write(struct.pack('<H', 24))
+    output_file.write(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')
+
+def _bytes_per_row(bitmap):
+    pixel_bytes = 3 * bitmap.width
+    padding_bytes = (4 - (pixel_bytes % 4)) % 4
+    return pixel_bytes + padding_bytes
+
+def _write_pixels(output_file, bitmap, palette):
+    # print('writing pixels')
+    row_buffer = bytearray(_bytes_per_row(bitmap))
+
+    for y in range(bitmap.height, 0, -1):
+        buffer_index = 0
+        for x in range(bitmap.width):
+            pixel = bitmap[x, y-1]
+            color = palette[pixel]
+            for _ in range(3):
+                row_buffer[buffer_index] = color & 0xFF
+                color >>= 8
+                buffer_index += 1
+        output_file.write(row_buffer)
+
+def save_bitmap(bitmap, palette, file_or_filename):
+    """Save a bitmap (using an associated palette) to a 24 bit per pixel BMP file.
+
+    :param bitmap: the displayio.Bitmap to save
+    :param palette: the displayio.Palette to use for looking up colors in the bitmap
+    """
+    print("Saving bitmap")
+    if not isinstance(bitmap, Bitmap):
+        raise ValueError('bitmap')
+    if not isinstance(palette, Palette):
+        raise ValueError('palette')
+    try:
+        if isinstance(file_or_filename, str):
+            output_file = open(file_or_filename, 'wb')
+        else:
+            output_file = file_or_filename
+
+        filesize = 54 + bitmap.height * _bytes_per_row(bitmap)
+        # print('Bitmap height: ', bitmap.height)
+        # print('Bitmap width: ', bitmap.width)
+        # print('Filesize: ', filesize)
+        _write_bmp_header(output_file, filesize)
+        _write_dib_header(output_file, bitmap)
+        _write_pixels(output_file, bitmap, palette)
+    except Exception:
+        print('Error saving bitmap')
```

Could you `raise` instead of printing here?